reference/hoon-expressions/rune/lus.md

```diff
@@ -135,42 +135,74 @@ Any Hoon expression, `q`, may be used to define the arm computation.
 
 ### +* "lustar"
 
-
-Produce a type constructor arm.
+Used to assign an alias to doors and define virtual arms.
 
 ##### Syntax
 
-Regular: **2-fixed**.
+Regular: **variadic**.
 
 ```hoon
-+*  p=term  [q=term]  r=spec
++*  a=term  b=hoon
+    c=term  d=hoon
+    ...
+    e=term  f=hoon
 ```
 
-`p` is the arm name, `q` is the name of the argument for the constructor function enclosed in square brackets, and `r` is a structure expression that defines the constructor function.
+`a`, `c`, `e` are arm names and `b`, `d`, `f` are any Hoon expression. Note that unlike all other
+runes with a variable number of arguments, the list of arguments of `+*` does
+not end with a terminating rune.
+
+`+*` arms must always come at the beginning of the battery, before any other
+type of lus arm.
 
 ##### Discussion
 
-Arms produced by `+*` are essentially type constructors.  They are used to construct new types from ones you already have.  Consider `list`, which takes some type -- e.g., `@`, `tape`, etc. -- and returns a new type, e.g., `(list @)`, `(list tape)`, etc.
+The primary use of `+*` is to assign aliases to doors (see Examples below).
+However, they may also be used to define "virtual arms" which behave just like
+ordinary arms except they allow nesting within cores that requires a specific
+number of arms, such as Gall apps. Put another way, no matter how many virtual
+arms are defined in a `+*` statement, the interpreter will see it as only being
+a single arm for the purposes of nesting.
 
-The Hoon subexpression, `r`, must be a structure expression.  That is, it must be either a basic structure expression (`*`, `~`, `^`, `?`, and `@`), or a complex expression made with the `$` family of runes (including irregular variants).
+Under the hood, `+*` gets compiled to `=*`.
 
 ##### Examples
 
+To assign an alias to a door, we often write the following.
+```hoon
+|_  foo
++*  this  .
 ```
-> =c |%
-       +*  triple  [a]  [a a a]
-       +*  wrap-flag  [a]  [? a]
-     --
+This is the idomatic way to assign the alias `this` to the door.
 
-> `(triple.c @)`[12 14 16]
-[12 14 16]
+Gall apps have a fixed number of arms, but sometimes you'd like to have more.
+This is where virtual arms come in.
 
-> `(triple.c ?)`[12 14 16]
-nest-fail
+```hoon
+|_  =bowl:gall
++*  this  .
+    samp  +<
+    cont  +>
+```
+This assigns the door the alias `this`, the sample of the door `samp`, and the
+context of the door `cont`.
 
-> `(triple.c ?)`[& | |]
-[%.y %.n %.n]
+Using more than two arguments for `+*` is unnecessary if you do not care about
+how many arms your core has. The above snippet has the same behavior as the
+following one does, except it has more arms:
 
-> `(wrap-flag.c @)`[& 22]
-[%.y 22]
-```
\ No newline at end of file
+```hoon
+|_  foo
++*  this  .
++*  samp  +<
++*  cont  +>
+```
+
+You are not restricted to lark expressions. Functions may be defined as well.
+Another common snipper seen in Gall apps is the following.
```

Comment:
We should avoid giving lark as examples, then. The `+*` is important here, we don't want people scratching their heads over its contents while trying to reason about the `+*` itself.

Reply:
Yes, good point!

Reply:
I've been trying to come up with a short and concise example of doing something other than assigning aliases with lustar, but I haven't quite been able to wrap my head around the examples I've come across in the code base.

Everywhere I've looked, you have something like

```
=<
  |_  =bowl:gall
  +*  do   ~(. +> bowl)
  ++  stuff  foo:do
::etc
--
|_  =bowl:gall
  ++  foo  %bar
::etc
--
```

I'm pretty sure I understand what is going on here - `do` is used to call arms in the second door with the sample of the first, but I don't understand the syntax with `%~`. The documentation says the syntax is `(arm door args)`, but `.` is the subject, which is the door, which isn't an arm...?

Are doors somehow considered to be arms, is the subject `.` is referring to not actually the parent door, is the doc on `%~` wrong, or what?

Reply:
IIUC `.` is in fact an arm; an arm is a subset of the subject, which can include the entire subject. So `~(. +> a)` is slamming `+>` (the door) with `a` to produce a core, and then evaluating `.` in the resulting expression, which is the core itself.